hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/matcher/SoftwareMouleJsonMatcher.java

```diff
@@ -0,0 +1,111 @@
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.matcher;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+import org.eclipse.hawkbit.repository.model.SoftwareModule;
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Factory;
+import org.hamcrest.Matcher;
+
+/**
+ * Set matcher for {@link SoftwareModule} and a list of
+ * {@link org.eclipse.hawkbit.dmf.json.model.SoftwareModule}.
+ */
+public final class SoftwareMouleJsonMatcher extends BaseMatcher<Set<SoftwareModule>> {
+
+    private final List<org.eclipse.hawkbit.dmf.json.model.SoftwareModule> expectedModules;
+
+    private SoftwareMouleJsonMatcher(List<org.eclipse.hawkbit.dmf.json.model.SoftwareModule> expectedModules) {
+        this.expectedModules = expectedModules;
+    }
+
+    /**
+     * Creates a matcher that matches when the list of repository software
+     * modules arelogically equal to the specified JSON software modules.
+     * <p>
+     * If the specified repository software modules are <code>null</code> then
+     * the created matcher will only match if the JSON software modules are
+     * <code>null</code>
+     * <p>
+     * For example:
+     * 
+     * <pre>
+     * List<SoftwareModule> modules;
+     * List<org.eclipse.hawkbit.dmf.json.model.SoftwareModule> expectedModules;
+     * 
+     * assertThat(modules, containsExactly(expectedModules));
+     * </pre>
+     * 
+     * @param expectedModules
+     *            the json sofware modules.
+     */
+    @Factory
+    public static Matcher<Set<SoftwareModule>> containsExactly(
+            final List<org.eclipse.hawkbit.dmf.json.model.SoftwareModule> expectedModules) {
+        return new SoftwareMouleJsonMatcher(expectedModules);
+    }
+
+    private static boolean containsExactly(Object actual,
+            List<org.eclipse.hawkbit.dmf.json.model.SoftwareModule> expected) {
+        if (actual == null) {
+            return expected == null;
+        }
+
+        @SuppressWarnings("unchecked")
+        final Collection<SoftwareModule> modules = (Collection<SoftwareModule>) actual;
+
+        if (modules.size() != expected.size()) {
+            return false;
+        }
+
+        for (final SoftwareModule repoSoftwareModule : modules) {
+            boolean containsElement = false;
+
+            for (final org.eclipse.hawkbit.dmf.json.model.SoftwareModule jsonSoftwareModule : expected) {
+                if (!jsonSoftwareModule.getModuleId().equals(repoSoftwareModule.getId())) {
+                    continue;
+                }
+                containsElement = true;
+
+                if (!jsonSoftwareModule.getModuleType().equals(repoSoftwareModule.getType().getKey())) {
```

Comment:
Can't you use a own matcher for `SoftwareModule`? So you can reuse it somewhere else